src/linear_solver_routines/block_col_projection.jl

```diff
@@ -0,0 +1,52 @@
+# This file is pat of RLinearAlgebra.jl
+# This was written by Nathaniel Pritchard
+"""
+    LinSysBlkColProj <: LinSysBlkColProjection
+
+A mutable structure that represents a standard block column projection method.
+
+# Aliases
+- `BlockCoordinateDescent`
+
+# Fields
+- `alpha::Float64`, a relaxation parameter that should be set between `0.0` and `2.0`
+- `G::GentData`, a buffer to store important information related to the Gentleman's 
+incremental QR least squares solver.
+
+Calling the constructor `LinSysBlkColProj()` defaults the relaxation parameter to `1.0`.
+
+"""
+mutable struct LinSysBlkColProj <: LinSysBlkColProjection 
+    α::Float64
+    G::Union{Nothing, GentData}
+    update::Union{Nothing, AbstractArray}
+end
+LinSysBlkColProj(α) = LinSysBlkColProj(α, nothing, nothing)
+
+LinSysBlkColProj() = LinSysBlkColProj(1.0, nothing, nothing)
+
+BlockCoordinateDescent = LinSysBlkColProj
+
+# Common rsubsolve interface for linear systems
+function rsubsolve!(
+    type::LinSysBlkColProj,
+    x::AbstractVector,
+    samp::Tuple{U,V,W,X} where {U<:Union{AbstractVector,AbstractMatrix},V<:AbstractArray,W<:AbstractVector,X<:AbstractVector},
+    iter::Int64,
+)
+    # samp[1] is the search direction
+    # samp[2] is the sketched matrix A
+    # samp[3] is the residual of system in A * samp[1], samp[1]' A' * (A * x - b)
+    # samp[4] is the residual of the system A * x - b
+    if iter == 1
+        m,p = size(samp[2])
+        rowBlockSize = m < 100000 ? m : min(div(m, 10), 100000)
+        type.G = Gent(samp[2], min(rowBlockSize, 10000))
+        type.update = Array{typeof(samp[2][1])}(undef, p)
+    end
```

Comment:
I agree I have added such comments. Basically, it runs a standard least squares solver, gentleman's with one row block, unless the matrix has more than 10,000 rows otherwise it runs gentleman's with the min of one 10th the number of rows and 10,000. This seemed to be roughly the point where there was benefits from using multiple blocks.

Reply:
Where are they?

Reply:
They should be located in the file; the location has changed to be inside of `linear_solver_routines/block_helper_functions`